mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RegistrationQueueName.java

```diff
@@ -0,0 +1,40 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class RegistrationQueueName {
+    private AtomicReference<String> queueName;
```

Comment:
Let's maybe use an Optional, that better wraps the `not existing concept`

Reply:
but Optional is a immutable object, once I set value to it, It cant be changed. While RabbitMQEventBus support continuously start & stop which leads to `RegistrationQueueName.queueName`  state changes from exist -> non exist and vice versa

Reply:
When we start and stop Event Bus multiple times, we can not reassign new value to `Optional` field

Reply:
Here we are clearly in a mutable context thus we can use assignation, like it is done in the `GroupRegistration`

Let's be consistent and keep using Optionals for not yet initialized stull.

(we can also add a precondition that the Optional is empty IMO)

Reply:
I changed it to  Optional, please check it